src/primitives.ts

```diff
@@ -955,76 +955,89 @@ export class PieChart {
     return normedRatio
   }
 
-  definePieParts(center: [number, number], radius: number): PiePart[] {
+  definePieParts(radius: number): PiePart[] {
     let initAngle: number = Math.PI/2;
     let nextAngle: number = initAngle;
+    const colorRatio: number = 360 / this.dataSamples.length;
+    let colorRadius: number = 0;
     let pieParts: PiePart[] = [];
 
-    let normedRatio = this.computeTotalSample();
+    let normedRatio = this.computeNormedRatio();
     this.dataSamples.forEach(sample => { 
+      colorRadius += colorRatio;
       nextAngle -= sample[this.slicingVariable] * normedRatio;
-      pieParts.push(new PiePart(center[0], center[1], radius, initAngle, nextAngle, true));
+
+      let newPart = new PiePart(radius, initAngle, nextAngle);
+      newPart.color = 'hsl('+ colorRadius +', 50%, 50%, 90%)';
+      pieParts.push(newPart);
+
       initAngle = nextAngle;
     })
     return pieParts
   }
 }
 
-
 export class PiePart {
   hidden_color: string = '';
-  isMouseOver: boolean = false;
-  isSelected: boolean = false;
   path: Path2D;
   clicked: boolean = false;
   color: string = '';
+  readonly overfliedColor: string = string_to_hex('lightskyblue');
+  readonly selectedColor: string = string_to_hex("red");
+  readonly center: [number, number] = [0, 0];
```

Comment:
I don't know if the PieChart or the PiePart should know the position of `center`.
I guess we could keep it here as it will be an easy change if it is needed.

However, you need to take advantage of center.x and center.y instead of center[0] and center[1].
If you don't want to implement Position and Coordinates for now, at least use something like this : 
`readonly center: Object = { x : 0, y : 0 };`

Reply:
I made an object in class definition.